src/OrbitGl/ThreadTrack.cpp

```diff
@@ -446,11 +446,17 @@ void ThreadTrack::OnCaptureComplete() {
 }
 
 [[nodiscard]] static inline uint64_t GetNextPixelBoundaryTimeNs(
-    float world_x, const internal::DrawData& draw_data) {
-  float normalized_x = (world_x - draw_data.track_start_x) / draw_data.track_width;
-  int pixel_x = static_cast<int>(
-      ceil(normalized_x * draw_data.viewport->WorldToScreenWidth(draw_data.track_width)));
-  return draw_data.min_tick + pixel_x * draw_data.ns_per_pixel;
+    uint64_t time_ns, const internal::DrawData& draw_data) {
+  uint64_t ns_from_start = time_ns - draw_data.min_tick;
+  uint64_t total_ns = draw_data.max_tick - draw_data.min_tick;
+  uint64_t num_pixels_on_track = draw_data.viewport->WorldToScreenWidth(draw_data.track_width);
+
+  // Making a ceiling with integers to not lose precision. Giving a maximum number of 4000 pixels
```

Comment:
"Given a track width of 4000 pixels, we can capture for more than a month without overflowing."

Orbit could span multiple screens, so it's better to mention the track width. Also, "giving" -> "given".

Reply:
Also, I had to validate that claim, my calculations bring me to ~53 days :-), I like the conservatism.

Reply:
I also arrived to the same number. I will use the days instead to be more precise.